Добалена API и документаия. Халанский М.Д. УВВ-112

MatveyKhalanski opened the pull request:
1.В API добавлены следующие методы
методы изменения координат курсора, изменение вида курсора,
 изменения размера курсора, метод гашения и восстановления курсора

2.Добавлена документация к этим методам
 
20. Реализовать класс Cursor. Полями являются координаты курсора по горизонтали и вертикали - целые положительные числа, вид курсора - горизонтальный или вертикальный, размер курсора - целое от 1 до 15. Реализовать методы изменения координат курсора, изменение вида курсора, изменения размера курсора, метод гашения и восстановления курсора

MatveyKhalanski committed:
Добалена API и документаия

1.В API добавлены следующие методы
методы изменения координат курсора, изменение вида курсора,
 изменения размера курсора, метод гашения и восстановления курсора

2.Добавлена документация к этим методам

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,58 @@
+*/
+class Cursor
+{
+public:
```
В Вашем случае, нужно реализовать приватную структуру `Point` с полями `x`, `y` типа [unsigned int](https://en.cppreference.com/w/cpp/language/types) и использвать ее для задания координат курсора

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,58 @@
+	int CoordX;
+	int CoordY;
+	bool CursorView;
+	int CursorSize;
```
здесь думать при реализации нужно будет. `int` точно не подойдет, может [unsigned char](https://en.cppreference.com/w/cpp/language/types)?
В конструкторе точно проверять на диапазон от `1..15`, который в `private` константами положить

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,58 @@
+/**
+* Класс Cursor
+*/
+class Cursor
```
моя ошибка. При объявлении скобки не нужны, нигде `{}`

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,58 @@
+	 * \brief Метод, возвращающий размер курсора от 1 до 15
+	 * \return размер курсора
+	 */
+	int GetCursorSize() const {};
```
тип метода изменить, когда изменить тип поля

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,58 @@
+	/**
+	 * \brief Метод, изменяющий координаты 
+	 */
+	Cursor& ChangeCoord(const Cursor& other) const {};
```
Вы меняете внутреннее состояние объекта...значит `const` точно нет!
Почему аргумент метода Другой курсор - не поняла, другие координаты?
Возвращает... тоже не поняла - другой курсор?
Из названия я подумала, что должно быть так 
`void ChangeCoord(const Point& newCoord)`;

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,58 @@
+	/**
+	 * \brief Метод, изменяющий вид курсора
+	 */
+	Cursor& ChangeCursorView(const Cursor& other) const {};
```
`const` неправильно, меняется внутреннее состояние
почему возвращаем `Cursor&`?

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,58 @@
+	/**
+	 * \brief Метод, изменяющий размер курсора 
+	 */
+	Cursor& ChangeCursorSize(const Cursor& other) const {};
```
см. предыдущие вопросы

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,58 @@
+	/**
+	 * \brief Метод гашения и востановлени¤ 
+	 */
+	Cursor& IsVisible(const Cursor& other) const {};
```
ответ - да или нет. Угадайте тип, возвращаемый методом)
Почему на вход подается что-то? Другой курсор? КАК?

MatveyKhalanski committed:
Исправление указаных ошибок #1

1. Убрал {} из Cursor.h
2. Вместо CoordX CoordY сделали структуру point с полями x y
3. Изменил тип поля CursorSize с int на unsigned char
4. Изменил тип метода GetCursorSize с int на unsigned char
5. Изменил тип метода ChangeCoord на void и передаю в него структуру
 point
6. Изменил тип метода ChangeCursorView на void и передаю в него Cursor
View
7. Изменил тип метода ChangeCursorSize на void и передаю в него Cursor
Size
8. Изменил тип метода IsVisible на bool и передаю в него Visible
9. Отредактировал методы GetCoordx и GetCoordy

MatveyKhalanski commented:
1. Убрал {} из Cursor.h
2. Вместо CoordX CoordY сделали структуру point с полями x y 
3. Изменил тип поля CursorSize с int на unsigned char
4. Изменил тип метода GetCursorSize с int на unsigned char
5. Изменил тип метода ChangeCoord на void и передаю в него структуру point 
6. Изменил тип метода ChangeCursorView на void и передаю в него CursorView
7. Изменил тип метода ChangeCursorSize на void и передаю в него CursorSize
8. Изменил тип метода IsVisible на bool и передаю в него Visible
9. Отредактировал методы GetCoordx и GetCoordy

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -5,54 +5,56 @@
-	Cursor(const int CoordX, const int CoordY, const bool CursorView, const int CursorSize) {};
-	Cursor(const Cursor& other) {};
+	Cursor();
+	Cursor(const unsigned int Point.x, const unsigned int Point.y, const bool CursorView, const int CursorSize);
```
а в чем плюс структуры?
```cpp
Cursor (const const unsigned int x, const const unsigned int x, const bool ..., const unsigned size);
Cursor (const Point& point, ...);

MarinaPaley replied:
про именование переменных типа `bool`: лучше имя начинать с глаголов
`is...`, `are...`, `has...`, которые будут понятны пользователям Вашей библиотеки.
В Вашем случае - `isVisible`

MarinaPaley replied:
У Вас не `bool`. Введите `class enum` с полями `horizontal` , `vertical`

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -5,54 +5,56 @@
-	Cursor() {};
-	Cursor(const int CoordX, const int CoordY, const bool CursorView, const int CursorSize) {};
-	Cursor(const Cursor& other) {};
+	Cursor();
```
нет его у Вас

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -5,54 +5,56 @@
 	 * \return горизонтальный или вертикальный
 	 */
-	bool GetCursorView() const {};
+	bool GetCursorView() const;
```
тип `enum` подойдет лучше

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -5,54 +5,56 @@
 	 * \return размер курсора
 	 */
-	int GetCursorSize() const {};
+	unsigned char GetCursorSize() const;
```
в конструкторе именно этот тип и надо принимать

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -5,54 +5,56 @@
 	 * \brief Метод, изменяющий вид курсора
 	 */
-	Cursor& ChangeCursorView(const Cursor& other) const {};
+	void ChangeCursorView(const CursorView& other);
```
что за ересь в качестве параметра?
после введения `enum` возвращаемым типом будет его имя

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -5,54 +5,56 @@
 	 * \brief Метод, изменяющий размер курсора 
 	 */
-	Cursor& ChangeCursorSize(const Cursor& other) const {};
+	void ChangeCursorSize(const CursorSize& other);
```
такое впечатление, что после написания некоторого количества правильных методов Вы просто устаете и пишите дичь(

MarinaPaley replied:
что это за типы такие `CursorSize&`, `CursorView&`

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -5,54 +5,56 @@
+	 * \brief Метод гашения и востановления 
 	 */
-	Cursor& IsVisible(const Cursor& other) const {};
+	bool IsVisible(const Visible& other);
```
почитайте что-нибудь про `C++`
в команде в тимс я хорошие книги положила

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -5,54 +5,56 @@
 	bool CursorView;
-	int CursorSize;
+	unsigned char CursorSize;
 	bool Visible;
```
`isVisible`

MarinaPaley reviewed (changes requested):
читая Ваш `API`, убеждаюсь, что Вы не понимаете процедурного программирования и типов данных. Почитайте книги об этом.

MatveyKhalanski committed:
Разработано API , реализация , документация #1

для  класса  Cursor

MatveyKhalanski committed:
Исправлен ряд ошибок #1

MarinaPaley commented on Lab1/Cursor.cpp:
```diff
@@ -0,0 +1,92 @@
+{
+	this->x = x;
+	this->y = y;
+	if (CursorSize<15){	
```
15 убрать в константу

MarinaPaley commented on Lab1/Cursor.cpp:
```diff
@@ -0,0 +1,92 @@
+	return this->y;
+}
+
+Orientation Cursor::isCursorView() const
```
GetOrientation

MarinaPaley replied:
Кому я написала??????
Что за имя метода???!?

MarinaPaley commented on Lab1/Cursor.cpp:
```diff
@@ -0,0 +1,92 @@
+	return this->visibility;
+}
+
+void Cursor::setCoord(unsigned int x, unsigned int y)
```
Не нужно

MarinaPaley reviewed (changes requested):
Сделайте метод toString и оператор сдвига

MatveyKhalanski committed:
Добавил константу и убрал лишний сеттер #2

MarinaPaley commented on Lab1/Cursor.cpp:
```diff
@@ -0,0 +1,86 @@
+
+void Cursor::setCursorSize(unsigned int size)
+{
+	if(size<=15){
```
Вы же ввели именованныб константу, почему 15?

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,78 @@
+class Cursor
+{
+public:
+	Cursor() = default;
```
Убрать!

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,78 @@
+{
+public:
+	Cursor() = default;
+	Cursor(const unsigned int x = 0, const unsigned int y = 0, const unsigned int CursorSize = 1, const bool visibility = true);
```
Добавить `explicit`

MarinaPaley commented on Lab1/Cursor.h:
```diff
@@ -0,0 +1,78 @@
+	 * \brief метод, возвращающий вид курсора
+	 * \return горизонтальный или вертикальный
+	 */
+	Orientation isCursorView() const;
```
`GetOrientation`

MarinaPaley commented:
переопределите оператор сдвига для вывода Вашего объекта (для зачета по Заданию2)

MatveyKhalanski committed:
Финальные исправления #2

MarinaPaley reviewed (approved):
1 - сдана, для 2 - переопределить оператор `>>`
